src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1338,13 +1337,19 @@ private static bool NormalizeTaskTypesCore(CSharpCompilation compilation, ref Ty
             switch (type.Kind)
             {
                 case SymbolKind.NamedType:
-                    var namedType = (NamedTypeSymbol)type;
-                    var changed = NormalizeTaskTypesCore(compilation, ref namedType);
-                    type = namedType;
-                    return changed;
+                    {
+                        var namedType = (NamedTypeSymbol)type;
+                        var changed = NormalizeTaskTypesCore(compilation, ref namedType);
+                        type = namedType;
+                        return changed;
+                    }
                 case SymbolKind.ArrayType:
-                    // PROTOTYPE(tasklike): Use VisitType or similar to cover all cases.
-                    break;
+                    {
```

Comment:
I think arrays, tuples and generics are the only constructed types. We'll have to make sure it works with Tuples. (Whose job is that?)